platforms/platforms.go

```diff
@@ -261,6 +261,5 @@ func Normalize(platform specs.Platform) specs.Platform {
 	// these fields are deprecated, remove them
 	platform.OSFeatures = nil
 	platform.OSVersion = ""
```

Comment:
This also needs to be removed. Otherwise, when we normalize the platform, the OSVersion gets dropped and the matcher will match all versions of Windows from a manifest (1809, ltsc2022, etc).

Reply:
thanks! I had this initially and dropped it during my updates. I can add it back here or rebase after  #6497